benchmarks/bench_plot_polynomial_kernel_approximation.py

```diff
@@ -0,0 +1,155 @@
+Benchmark for explicit feature map approximation of polynomial kernels
+========================================================================
+
+An example illustrating the approximation of the feature map
+of an Homogeneous Polynomial kernel.
+
+.. currentmodule:: sklearn.kernel_approximation
+
+It shows how to use :class:`PolynomialSampler` and :class:`Nystroem` to
+approximate the feature map of a polynomial kernel for
+classification with an SVM on the digits dataset. Results using a linear
+SVM in the original space, a linear SVM using the approximate mappings
+and a kernelized SVM are compared.
+
+The first plot shows the classification accuracy of Nystroem [2] and
+PolynomialSampler [1] as the output dimension (n_components) grows.
+It also shows the accuracy of a linear SVM and a polynomial kernel SVM
+on the same data.
+
+The second plot explores the scalability of PolynomialSampler and Nystroem.
+For a sufficiently large output dimension, PolynomialSampler should be faster
+as it is O(n(d+klog k)) while Nystroem is O(n(dk+k^2)). In addition,
+Nystroem requires a time-consuming training phase, while training is
+almost immediate for PolynomialSampler, whose training phase boils down to
+initializing some random variables (because is data-independent).
+
+[1] Pham, N., & Pagh, R. (2013, August). Fast and scalable polynomial
+kernels via explicit feature maps. In Proceedings of the 19th ACM SIGKDD
+international conference on Knowledge discovery and data mining (pp. 239-247)
+(http://chbrown.github.io/kdd-2013-usb/kdd/p239.pdf)
+
+[2] Charikar, M., Chen, K., & Farach-Colton, M. (2002, July). Finding frequent
+items in data streams. In International Colloquium on Automata, Languages, and
+Programming (pp. 693-703). Springer, Berlin, Heidelberg.
+(http://www.vldb.org/pvldb/1/1454225.pdf)
+
+"""
+# Author: Daniel Lopez-Sanchez <lope@usal.es>
+# License: BSD 3 clause
+
+# Load data manipulation functions
+from sklearn.datasets import load_digits
+from sklearn.model_selection import train_test_split
+
+# Some common libraries
+import matplotlib.pyplot as plt
+import numpy as np
+
+# Will use this for timing results
+from time import time
+
+# Import SVM classifiers and feature map approximation algorithms
+from sklearn.svm import LinearSVC, SVC
+from sklearn.kernel_approximation import Nystroem, PolynomialSampler
+from sklearn.pipeline import Pipeline
+
+# Split data in train and test sets
+X, Y = load_digits()["data"], load_digits()["target"]
+X_train, X_test, Y_train, Y_test = train_test_split(X, Y, train_size=0.7)
+
+# Set the range of n_components for our experiments
+out_dims = range(20, 400, 20)
+
+# Evaluate Linear SVM
+lsvm = LinearSVC().fit(X_train, Y_train)
+lsvm_score = 100*lsvm.score(X_test, Y_test)
+
+# Evaluate kernelized SVM
+ksvm = SVC(kernel="poly", degree=2, gamma=1.).fit(X_train, Y_train)
+ksvm_score = 100*ksvm.score(X_test, Y_test)
+
+# Evaluate PolynomialSampler + LinearSVM
+ps_svm_scores = []
+n_test = 5
```

Comment:
Better to call this `n_runs` as  "test" is a already quite overloaded term in CS and ML contexts.

Reply:
Fixed.